model/src/main/kotlin/VulnerabilityReference.kt

```diff
@@ -52,14 +52,15 @@ data class VulnerabilityReference(
 ) {
     companion object {
         /**
-         * Return a human-readable string that is determined based on the given score and scoring system.
+         * Return a human-readable string that is determined based on the given [scoringSystem] and [severity].
          */
-        fun getSeverityString(scoringSystem: String?, severity: String) =
-            when (scoringSystem) {
-                "CVSS2" -> Cvss2Rating.fromScore(severity.toFloat())?.name?.lowercase()
-                "CVSS3" -> Cvss3Rating.fromScore(severity.toFloat())?.name?.lowercase()
-                else -> "unknown"
-            }
+        @Suppress("UNUSED") // This function is used in the templates.
+        fun getSeverityString(scoringSystem: String?, severity: String?) =
+            when (scoringSystem?.uppercase()) {
+                "CVSS2", "CVSSV2" -> severity?.toFloat()?.let { Cvss2Rating.fromScore(it)?.toString() }
```

Comment:
As we do not have control over the quality of data returned by a vulnerability provider, I would prefer a more robust implementation for the conversion to a numeric value. `toFloat()` may just blow up for unexpected input. Maybe have an extension function `toScore()` on `String` that catches `NumberFormatException`?

Reply:
I'm simply using `toFloatOrNull()` now.